### PR TITLE
[LPT] Concat complex graph support

### PR DESCRIPTION
**Pull request by eshoguli**
issues:
- 28042: Support by run-time in IE MKLDNN plugin for Human-pose (single, 3d) INT8 model  
- 28043: Support by run-time in IE CLDNN plugin for Human-pose (single, 3d) INT8 model  
- 31797: Quantized model with per-channel quantization for element-wise operations has zero accuracy

### CODE REVIEW

**Comment by eshoguli** [comment]
@dmitry-gorokhov, please review and comment.

File: inference-engine/tests/functional/plugin/shared/src/low_precision_transformations/concat_neighboring_graph_transformation.cpp [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    SetRefMode(LayerTestsUtils::RefMode::IE);
+
+    InferenceEngine::SizeVector inputShape;
+    InferenceEngine::Precision netPrecision;
+    InferenceEngine::details::LayerTransformation::Params params;
+    std::tie(netPrecision, inputShape, targetDevice, params) = this->GetParam();
+    auto ngPrecision = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrecision);
+
+    const auto paramNode1 = std::make_shared<ngraph::opset1::Parameter>(ngPrecision, ngraph::Shape(inputShape));
+    const auto fakeQuantize1 = makeFakeQuantize(ngPrecision, paramNode1->output(0));
```
Comment (dmitry-gorokhov) [inline inference-engine/tests/functional/plugin/shared/src/low_precision_transformations/concat_neighboring_graph_transformation.cpp:46]: ngraph builders contain its own makeFQ routines: https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/tests/ngraph_functions/include/ngraph_functions/builders.hpp#L168. Is it possible to reuse?

File: inference-engine/tests/functional/plugin/shared/src/low_precision_transformations/concat_neighboring_graph_transformation.cpp [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+    // check quantized FQ layers map: should includes all FQ
+
+    IE_SUPPRESS_DEPRECATED_START
+}
+
+TEST_P(ConcatNeighboringGraphTransformation, CompareWithRefImpl) {
+    Run();
+
+    if (targetDevice == std::string{CommonTestUtils::DEVICE_GPU}) {
```
Comment (dmitry-gorokhov) [inline inference-engine/tests/functional/plugin/shared/src/low_precision_transformations/concat_neighboring_graph_transformation.cpp:105]: Not necessary to put this condition anymore. It was moved into base class.

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/low_precision_transformations/concat_neighboring_graph_transformation.cpp [hunk 3]
```diff
@@ -0,0 +1,36 @@
+// Copyright (C) 2019 Intel Corporation
```
[the remark this answers is not in the document]
Comment (eshoguli) [reply inference-engine/tests/functional/plugin/cpu/shared_tests_instances/low_precision_transformations/concat_neighboring_graph_transformation.cpp]: done for all affected files in PR, thanks!

File: inference-engine/tests/functional/plugin/gpu/shared_tests_instances/low_precision_transformations/concat_transformation.cpp [hunk 4]
```diff
@@ -8,17 +8,29 @@
...
 namespace {
 const std::vector<InferenceEngine::Precision> netPrecisions = {
         InferenceEngine::Precision::FP32,
         InferenceEngine::Precision::FP16
 };
 
+const std::vector<LayerTransformation::Params> trasformationParamValues = {
+    LayerTestsUtils::LayerTransformationParamsFactory::createParamCpu(),
+    LayerTestsUtils::LayerTransformationParamsFactory::createParamI8I8(),
+    LayerTestsUtils::LayerTransformationParamsFactory::createParamU8I8()
```
[the remark this answers is not in the document]
Comment (eshoguli) [reply inference-engine/tests/functional/plugin/gpu/shared_tests_instances/low_precision_transformations/concat_transformation.cpp]: done, for GPU tests, thanks!

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/low_precision_transformations/concat_neighboring_graph_transformation.cpp [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+INSTANTIATE_TEST_CASE_P(LPT, ConcatNeighboringGraphTransformation,
+    ::testing::Combine(
+        ::testing::ValuesIn(netPrecisions),
+        ::testing::Values(InferenceEngine::SizeVector({ 1, 1024, 16, 16 })),
+        ::testing::Values(CommonTestUtils::DEVICE_CPU),
+        ::testing::ValuesIn(trasformationParamValues)),
+    ConcatNeighboringGraphTransformation::getTestCaseName);
+}  // namespace
+
```
[the remark this answers is not in the document]
Comment (eshoguli) [reply inference-engine/tests/functional/plugin/cpu/shared_tests_instances/low_precision_transformations/concat_neighboring_graph_transformation.cpp]: done for all affected files, thanks!

File: inference-engine/tests/functional/plugin/shared/src/low_precision_transformations/concat_neighboring_graph_transformation.cpp [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+    InferenceEngine::details::LayerTransformation::Params params;
+    std::tie(netPrecision, inputShapes, targetDevice, params) = obj.param;
+
+    std::ostringstream result;
+    result << netPrecision.name() << "_" << targetDevice << "_" << toString(params);
+    return result.str();
+}
+
+void ConcatNeighboringGraphTransformation::SetUp() {
+    SetRefMode(LayerTestsUtils::RefMode::IE);
```
[the remark this answers is not in the document]
Comment (eshoguli) [reply inference-engine/tests/functional/plugin/shared/src/low_precision_transformations/concat_neighboring_graph_transformation.cpp]: Fixed for both cases. Note, please, Interpolation operation inference is not supported bu nGraph function. As result limitation is still used in one case.

File: inference-engine/tests/functional/plugin/shared/src/low_precision_transformations/concat_neighboring_graph_transformation.cpp (hunk 1 above)
Comment (eshoguli) [reply inference-engine/tests/functional/plugin/shared/src/low_precision_transformations/concat_neighboring_graph_transformation.cpp]: Reused, base test class was updated. thanks!

File: inference-engine/tests/functional/plugin/shared/src/low_precision_transformations/concat_neighboring_graph_transformation.cpp (hunk 2 above)
Comment (eshoguli) [reply inference-engine/tests/functional/plugin/shared/src/low_precision_transformations/concat_neighboring_graph_transformation.cpp]: remove, thanks!

**Comment by dmitry-gorokhov** [comment]
@eshoguli  I see that one of the ConcatNeighboringGraphTransformation tests failed. Looks like sporadic caused by randomized FQ values. If this is true lets get rid of random values.

File: inference-engine/tests/ie_test_utils/functional_test_utils/low_precision_transformations/layer_transformation.cpp [hunk 7]
```diff
@@ -23,43 +23,70 @@
...
-            true,  // quantizeOutputs
-            true,  // weightsToConst
-            InferenceEngine::details::LayerTransformation::QuantizedTensorAlignment::UpdateLevel,  // quantizedTensorAlignmentOnActivations
-            InferenceEngine::details::LayerTransformation::QuantizedTensorAlignment::None,  // quantizedTensorAlignmentOnWeights
-            true,  // roundQuantizedValues
-            true,  // updateBiases
-            true); // supportAsymmetricQuantization
-        auto transformer = getLowPrecisionTransformer(params);
-        transformer.transform(*cnnNetworkImp);
+InferenceEngine::details::LayerTransformation::Params LayerTransformationParamsFactory::createParamCpu() {
```
Comment (iefode) [inline inference-engine/tests/ie_test_utils/functional_test_utils/low_precision_transformations/layer_transformation.cpp:68]: As I can see, It's code duplication. And more, I think it's bad idea to add parameters per HW

File: inference-engine/tests/ie_test_utils/functional_test_utils/low_precision_transformations/layer_transformation.hpp [hunk 8]
```diff
@@ -7,17 +7,84 @@
...
+    static InferenceEngine::details::LayerTransformation::Params createParamCpu();
+    static InferenceEngine::details::LayerTransformation::Params createParamGpu();
+};
+
+template <typename T>
+class LayerTransformation : public testing::WithParamInterface<T>, public LayerTestsUtils::LayerTestsCommon {
 public:
     InferenceEngine::details::LowPrecisionTransformations getLowPrecisionTransformations(
-        const InferenceEngine::details::LayerTransformation::Params& params) const;
+        const InferenceEngine::details::LayerTransformation::Params& params) const {
```
Comment (iefode) [inline inference-engine/tests/ie_test_utils/functional_test_utils/low_precision_transformations/layer_transformation.hpp:35]: Please move function definition to the cpp file

File: inference-engine/tests/ie_test_utils/functional_test_utils/low_precision_transformations/layer_transformation.hpp [hunk 9]
```diff
@@ -7,17 +7,84 @@
...
+        if (targetDevice == "CPU") {
```
Comment (iefode) [inline inference-engine/tests/ie_test_utils/functional_test_utils/low_precision_transformations/layer_transformation.hpp:36]: Why we use device separation in common utils lib?

File: inference-engine/tests/ie_test_utils/functional_test_utils/low_precision_transformations/layer_transformation.hpp [hunk 10]
```diff
@@ -7,17 +7,84 @@
...
+        } else if (targetDevice == "GPU") {
+            return InferenceEngine::details::LowPrecisionTransformer::getAllTransformations(params);
+        } else {
+            THROW_IE_EXCEPTION << "unknown target device " << targetDevice;
+        }
+    }
 
     InferenceEngine::details::LowPrecisionTransformer getLowPrecisionTransformer(
-        const InferenceEngine::details::LayerTransformation::Params& params) const;
+        const InferenceEngine::details::LayerTransformation::Params& params) const {
```
Comment (iefode) [inline inference-engine/tests/ie_test_utils/functional_test_utils/low_precision_transformations/layer_transformation.hpp:51]: Please move function definition to the cpp file

File: inference-engine/tests/ie_test_utils/functional_test_utils/low_precision_transformations/layer_transformation.cpp (hunk 7 above)
Comment (eshoguli) [reply inference-engine/tests/ie_test_utils/functional_test_utils/low_precision_transformations/layer_transformation.cpp:68]: good comment, thanks: 32278: [LPT] [TEST] Move hardware specific logic to parent classes

File: inference-engine/tests/ie_test_utils/functional_test_utils/low_precision_transformations/layer_transformation.hpp (hunk 9 above)
Comment (eshoguli) [reply inference-engine/tests/ie_test_utils/functional_test_utils/low_precision_transformations/layer_transformation.hpp:36]: good comment, thanks: 32278: [LPT] [TEST] Move hardware specific logic to parent classes

**Comment by iefode** [comment]
Approve with TODO: Fix ISSUE: 32276.

**Pull request merged**